### PR TITLE
COL-914, record activities pin, repin, get_pin, get_repin

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-914

* pinning, under the hood, is identified as first-time or a repeat (re-pin)
* actor_id is needed in get_pin / get_repin logic
* helper functions moved to  `col-activities/lib/util.js`

### CODE REVIEW

File: node_modules/col-activities/lib/api.js [hunk 1]
```diff
@@ -1006,144 +838,84 @@ var recalculateImpactScores = module.exports.recalculateImpactScores = function(
...
-        log.info('Recalculation of trending scores complete');
 
         return callback();
       });
     });
-  });
+
+  } else {
+    return callback();
+  }
```
Comment (johncrossman) [inline node_modules/col-activities/lib/api.js]: snippet above is a bit clearer w/ `?w=1`

File: node_modules/col-activities/tests/test-points.js [hunk 2]
```diff
@@ -301,6 +302,98 @@ describe('Activity Points', function() {
...
+      // The asset_users are user1 and user2; the asset has been pinned by user2 and user3.
+      AssetsTestUtil.createPinningScenario(function(client1, client2, client3, course, user1, user2, user3, asset) {
+
+        AssetsTestUtil.getDbUser(user1.id, function(dbUser1) {
+          AssetsTestUtil.getDbUser(user2.id, function(dbUser2) {
+            AssetsTestUtil.getDbUser(user3.id, function(dbUser3) {
+
+              // Verify appropriate activity counts
+              ActivitiesTestUtil.assertGetActivitiesForUserId(client1, course, dbUser1.id, function(activities1) {
+                // user1 received a two pins
```
Comment (paulkerschen) [inline node_modules/col-activities/tests/test-points.js]: @johncrossman `a two pins`

Comment (johncrossman) [inline node_modules/col-activities/tests/test-points.js]: Yes, I like the lilt of it. I'll commit the amend once you're done with review.

**Comment by paulkerschen** [comment]
@johncrossman This all looks really good - especial thanks for readable tests. I think it's good to merge after typo fix.

**Comment by johncrossman** [comment]
Amendment is in. Thx, @paulkerschen

**Pull request merged**